public/app/features/dashboard/services/SnapshotSrv.ts

```diff
@@ -0,0 +1,31 @@
+import { backendSrv } from 'app/core/services/backend_srv';
+import { DashboardDTO } from 'app/types';
+
+// Used in the snapshot list
+export interface Snapshot {
+  key: string;
+  name: string;
+  external: boolean;
+  externalUrl?: string;
+  url?: string;
+}
+
+export interface DashboardSnapshotSrv {
+  getSnapshots: () => Promise<Snapshot[]>;
+  deleteSnapshot: (key: string) => Promise<void>;
+  getSnapshot: (key: string) => Promise<DashboardDTO>;
+}
+
+const legacyDashboardSnapshotSrv: DashboardSnapshotSrv = {
+  getSnapshots: () => backendSrv.get<Snapshot[]>('/api/dashboard/snapshots'),
+  deleteSnapshot: (key: string) => backendSrv.delete('/api/snapshots/' + key),
+  getSnapshot: async (key: string) => {
+    const dto = await backendSrv.get<DashboardDTO>('/api/snapshots/' + key);
+    dto.meta.canShare = false;
```

Comment:
^^^ the key change

Reply:
ya -- my first version of the PR added it, but that starts the cascade of updating all the swager docs etc